internal/service/alarms/internal/db/migrations/000003_create_alarm_event_record_table.up.sql

```diff
@@ -39,21 +39,33 @@ CREATE TABLE IF NOT EXISTS alarm_event_record (
 -- Set ownership of the alarm_sequence_seq sequence to alarm_event_record.alarm_sequence_number
 ALTER SEQUENCE alarm_sequence_seq OWNED BY alarm_event_record.alarm_sequence_number;
 
--- Function to update the alarm_sequence_number on specific status or time changes
-CREATE OR REPLACE FUNCTION update_alarm_event_sequence()
+-- Function to execute before updating alarm_event_record
+CREATE OR REPLACE FUNCTION update_alarm_event_record()
     RETURNS TRIGGER AS $$
 BEGIN
+    -- Update alarm_acknowledged_time if alarm_acknowledged is set to TRUE
```

Comment:
Is there a specific reason for doing this work in a trigger?   This can easily be handled while setting the acknowledged attribute by the server.   Triggers are a pain to maintain and not portable so I'd prefer to limit our use of them to a minimum.

Reply:
Yeah, let's keep it in code. Thanks!